.github/workflows/build.yml

```diff
@@ -0,0 +1,68 @@
+name: build
+
+on:
+  push:
+    branches:
+      - master
+      - staging
+      - develop
+      - sandbox
+
+jobs:
+  get-environment-name:
+    name: "Extract environment name"
+    runs-on: ubuntu-latest
+    outputs:
+      environment: ${{ steps.extract.outputs.environment }}
+    steps:
+      - id: extract
+        run: echo "::set-output name=environment::$(echo $GITHUB_REF | sed -e '/^refs\/heads\/\(.*\)$/!d;s//\1/')"
+      - env:
+          EXTRACTED: ${{ steps.extract.outputs.environment }}
+        run: 'echo "Extracted environment name: $EXTRACTED"'
+
+  build:
+    runs-on: ubuntu-latest
+    needs:
+      - get-environment-name
+    environment: ${{ needs.get-environment-name.outputs.environment }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          submodules: true
+
+      - name: Auth gcloud
+        uses: 'google-github-actions/auth@v0'
+        with:
+          credentials_json: ${{ secrets.GCP_SERVICE_CREDENTIALS }}
+
+      - name: Setup gcloud
+        uses: 'google-github-actions/setup-gcloud@v0'
+
+      - name: Configure docker
+        run: gcloud auth configure-docker ${{ secrets.GCP_REGISTRY_URL }}
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Inspect builder
+        run: |
+          echo "Name:      ${{ steps.buildx.outputs.name }}"
+          echo "Endpoint:  ${{ steps.buildx.outputs.endpoint }}"
+          echo "Status:    ${{ steps.buildx.outputs.status }}"
+          echo "Flags:     ${{ steps.buildx.outputs.flags }}"
+          echo "Platforms: ${{ steps.buildx.outputs.platforms }}"
+
+      - name: Build and push
+        run: |
+          docker buildx build \
+          -f docker/server/Dockerfile . \
+          --platform linux/amd64,linux/arm64 \
+          --tag ${{ secrets.GCP_REGISTRY_URL }}/${{ secrets.GCP_PROJECT_ID }}/${{ secrets.GCP_REGISTRY_REPO }}/${{ secrets.GCP_APP_NAME }}:${{ github.sha }} \
+          --tag ${{ secrets.GCP_REGISTRY_URL }}/${{ secrets.GCP_PROJECT_ID }}/${{ secrets.GCP_REGISTRY_REPO }}/${{ secrets.GCP_APP_NAME }}:latest \
+          --push
```

Comment:
Our applications are currently using some existing actions for GCP auth + docker buildx commands:

```
    - name: Login to GAR
      uses: docker/login-action@v1
      with:
        registry: us-central1-docker.pkg.dev/${{ inputs.gcp_project_id }}
        username: _json_key
        password: ${{ inputs.docker_artifact_password }}
    - name: Build and push
      uses: docker/build-push-action@v2
      with:
        context: .
        platforms: linux/amd64
        push: true
        tags: |
          us-central1-docker.pkg.dev/${{ inputs.gcp_project_id }}/planning-service-images/planning-service:${{ inputs.namespace }}
          us-central1-docker.pkg.dev/${{ inputs.gcp_project_id }}/planning-service-images/planning-service:${{ inputs.commit_sha }}
        build-args: |
          NPM_TOKEN=${{ inputs.npm_token }}
```

May be cleaner and more consistent to our other approaches